src/localkiwixserver.cpp

```diff
@@ -84,18 +88,24 @@ void LocalKiwixServer::runOrStopServer()
         auto settingsManager = KiwixApp::instance()->getSettingsManager();
         m_port = ui->PortChooser->text().toInt();
         mp_server->setPort(m_port);
-        m_ipAddress = (ui->IpChooser->currentText() != gt("all")) ? ui->IpChooser->currentText() : "0.0.0.0";
         settingsManager->setKiwixServerPort(m_port);
-        settingsManager->setKiwixServerIpAddress(m_ipAddress);
-        mp_server->setAddress(m_ipAddress.toStdString());
-        m_ipAddress = (m_ipAddress != "0.0.0.0") ? m_ipAddress : QString::fromStdString(kiwix::getBestPublicIp());
-        ui->IpAddress->setText("http://" + m_ipAddress + ":" + QString::number(m_port));
-        ui->IpAddress->setReadOnly(true);
+        mp_server->setIpMode(kiwix::IpMode::AUTO);
+        mp_server->setAddress("");
+        (ui->IpChooser->currentText() == gt("all_ips")) ? mp_server->setIpMode(kiwix::IpMode::ALL)  :
+        (ui->IpChooser->currentText() == gt("ipv4"))    ? mp_server->setIpMode(kiwix::IpMode::IPV4) :
+        (ui->IpChooser->currentText() == gt("ipv6"))    ? mp_server->setIpMode(kiwix::IpMode::IPV6) :
+                                                          mp_server->setAddress(ui->IpChooser->currentText().toStdString());
```

Comment:
Conditional expressions shouldn't be used in place of statements, i.e. when you care about the side-effect rather than the value of the expression. Change to `if() {} else if () {} ...`